src/pyodide/internal/workers.py

```diff
@@ -607,6 +640,89 @@ async def text(self) -> str:
         return await self.js_object.text()
 
 
+def to_python(obj: "JsProxy", throw=True):
+    """
+    Converts JS objects like Response, Request, Blob, etc. to equivalent Python objects defined in
+    this module and also other JS objects like Map, Set, etc. to equivalent Python stdlib objects.
+    """
+    if not hasattr(obj, "constructor"):
+        if throw:
+            raise TypeError(
+                f"Expected a JS object with a `constructor` field, got {type(obj).__name__}"
+            )
+        else:
+            return obj
+
+    if obj.constructor.name == "Response":
+        return Response(obj)
+    elif obj.constructor.name == "FormData":
+        return FormData(obj)
+    elif obj.constructor.name == "Blob":
+        return Blob(obj)
+    elif obj.constructor.name == "File":
+        return File(obj)
+    elif obj.constructor.name == "Request":
+        return Request(obj)
+    elif obj.constructor.name == "Date":
+        # XXX: Pyodide should gain support for this, we should upstream this.
+        return datetime.datetime.fromtimestamp(obj.getTime() / 1000)
+
+    return (
+        obj.to_py()
+    )  # TODO: This can fail, we should check if it returned a JsProxy and throw.
+
+    if throw:
+        raise TypeError(f"Unsupported JS object, got {obj.constructor.name}")
+    else:
+        return obj
+
+
+def convert_to_js(obj) -> JsProxy:
+    if obj is None:
+        return obj
+
+    if hasattr(obj, "js_object"):
+        return obj.js_object
+
+    if isinstance(obj, datetime.datetime):
+        # XXX: Pyodide should gain support for this, we should upstream this.
+        return js.Date.new(obj.timestamp() * 1000)
+
+    if isinstance(obj, (tuple, bytearray)):
+        raise TypeError(f"DataCloneError: {type(obj)} cannot be sent over RPC.")
```

Comment:
This only works for the outermost layer unfortunately. It will still convert `[(1,2)]` to an array of arrays.

Reply:
Nope. I probably should add a way to adjust the conversion of stuff before we apply the default logic to it.